config/v1/types_infrastructure.go

```diff
@@ -25,15 +25,31 @@ type InfrastructureSpec struct {
 }
 
 type InfrastructureStatus struct {
-	// cloudProvider is the IaaS provider that is running the cluster.
-	//
-	// Valid values are:
-	// - aws
-	// - openstack
-	// +optional
-	CloudProvider string `json:"cloudProvider,omitempty"`
+	// platform is the underlying infrastructure provider for the cluster. This
+	// value controls whether infrastructure automation such as service load
+	// balancers, dynamic volume provisioning, machine creation and deletion, and
+	// other integrations are enabled. If None, no infrastructure automation is
+	// enabled.
+	Platform PlatformType `json:"platform,omitempty"`
 }
 
+// platformType is a specific supported infrastructure provider.
+type PlatformType string
+
+const (
+	// awsPlatform represents Amazon AWS.
+	AWSPlatform PlatformType = "aws"
+
+	// openStackPlatform represents OpenStack.
+	OpenStackPlatform PlatformType = "openstack"
+
+	// libvirtPlatform represents libvirt.
+	LibvirtPlatform PlatformType = "libvirt"
+
+	// nonePlatform means there is no infrastructure provider.
+	NonePlatform PlatformType = "none"
```

Comment:
We should be using Kube style constants unless we have a reason not to.  `None`, `LibVirt`, `AWS`, `OpenStack`

Reply:
Fixed, thanks for the feedback